go/docs/github.go

```diff
@@ -0,0 +1,218 @@
+				DownloadUrl string
+			}
+		} `graphql:"releaseAssets(first: 20)"`
+	}
+
+	var releasesQuery struct {
+		Repository struct {
+			Releases struct {
+				Nodes []GitHubRelease
+			} `graphql:"releases(first: 10, orderBy: {field: CREATED_AT, direction: DESC})"`
+		} `graphql:"repository(owner: \"concourse\", name: \"concourse\")"`
+	}
+
+	err := client.Query(ctx, &releasesQuery, nil)
+	if err != nil {
+		return nil, fmt.Errorf("fetch releases: %w", err)
+	}
+
+	releases := releasesQuery.Repository.Releases.Nodes
+
+	var latestRelease *GitHubRelease
+	var latestVersion semver.Version
+	if len(releases) == 0 {
+		return nil, errors.New("no releases found!")
+	}
+	for _, release := range releases {
+		if release.IsPrerelease {
+			continue
+		}
+
+		if release.TagName == nil {
+			return nil, fmt.Errorf("no tag name for release %v", release)
+		}
+		version, err := semver.ParseTolerant(*release.TagName)
+		if err != nil {
+			return nil, err
+		}
+
+		if latestRelease == nil || version.GT(latestVersion) {
+			latestRelease = &release
+			latestVersion = version
+		}
+	}
+
+	var linuxURL, darwinURL, windowsURL string
+	var flyLinuxURL, flyDarwinURL, flyWindowsURL string
+	for _, asset := range latestRelease.Assets.Nodes {
+		name := asset.Name
+
+		if strings.Contains(name, "concourse") && strings.Contains(name, "linux") && strings.HasSuffix(name, ".tgz") {
+			linuxURL = asset.DownloadUrl
+		}
+
+		if strings.Contains(name, "concourse") && strings.Contains(name, "darwin") && strings.HasSuffix(name, ".tgz") {
+			darwinURL = asset.DownloadUrl
+		}
+
+		if strings.Contains(name, "concourse") && strings.Contains(name, "windows") && strings.HasSuffix(name, ".zip") {
+			windowsURL = asset.DownloadUrl
+		}
+
+		if strings.Contains(name, "fly") && strings.Contains(name, "linux") && strings.HasSuffix(name, ".tgz") {
+			flyLinuxURL = asset.DownloadUrl
+		}
+
+		if strings.Contains(name, "fly") && strings.Contains(name, "darwin") && strings.HasSuffix(name, ".tgz") {
+			flyDarwinURL = asset.DownloadUrl
+		}
+
+		if strings.Contains(name, "fly") && strings.Contains(name, "windows") && strings.HasSuffix(name, ".zip") {
+			flyWindowsURL = asset.DownloadUrl
+		}
+	}
+
+	return booklit.Styled{
+		Style: "download-links",
+		Block: true,
+		Content: booklit.Link{
+			Target:  latestRelease.Url,
+			Content: booklit.String(latestRelease.Name),
+		},
+		Partials: booklit.Partials{
+			"Version":         booklit.String(latestRelease.Name),
+			"ReleaseNotesURL": booklit.String(latestRelease.Url),
+			"LinuxURL":        booklit.String(linuxURL),
+			"DarwinURL":       booklit.String(darwinURL),
+			"WindowsURL":      booklit.String(windowsURL),
+			"FlyLinuxURL":     booklit.String(flyLinuxURL),
+			"FlyDarwinURL":    booklit.String(flyDarwinURL),
+			"FlyWindowsURL":   booklit.String(flyWindowsURL),
+		},
+	}, nil
+}
+
+func (p *Plugin) githubClient(ctx context.Context) *githubv4.Client {
+	var tc *http.Client
+	var githubToken = os.Getenv("GITHUB_TOKEN")
```

Comment:
Not much you can really do about this without complicating the code a fair bit, but kind of a bummer that this means you need to set `GITHUB_TOKEN` whenever you want to run the docs locally

Reply:
ah yeah true, let me change this to fall back on filler data when the token isn't set

Reply:
done!